javagen/src/main/java/com/azure/autorest/template/ConvenienceMethodTemplateBase.java

```diff
@@ -138,6 +151,46 @@ protected void writeMethodImplementation(
         writeInvocationAndConversion(convenienceMethod, protocolMethod, invocationExpression, methodBlock, typeReferenceStaticClasses);
     }
 
+    private static void writeParameterTransformation(MethodTransformationDetail detail, ClientMethod convenienceMethod, JavaBlock methodBlock) {
```

Comment:
Long term, `MethodTransformationDetail` might not be the best way to handle the flatten/grouping (or something more advanced).